firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt

```diff
@@ -41,10 +40,10 @@ class FirebaseSessions internal constructor(firebaseApp: FirebaseApp) {
   fun greeting(): String = "Matt says hi!"
 
   private fun initiateSessionStart() {
-    // TODO(mrober): Generate a session
-    Log.i(TAG, "Initiate session start")
+    val sessionState = sessionGenerator.generateNewSession()
+    val sessionEvent = SessionEvent.sessionStart(sessionState)
```

Comment:
I think this is very close to the iOS implementation. Just to confirm, here is how I think the current piece of code would evolve for event dispatch. Please let me know if you think otherwise.

1. Session Generator is stateful and it returns sessions related information for the current session event. (ID, Index, previous ID, parent ID and likes)
2.SessionEvent class takes those details to fill in more details
3. Firebase Sessions later embellishes the event with details like data collection state.
4. Firebase Sessions will use Firelog services to hand over the event.

Curious as to where would the config layer fit in here and how would sampling be done to either capture or skip an event?

Reply:
Yup, and for sampling we'll handle that when we do Settings. We will pass in something like `collectEvents = shouldCollectEvents(...)` to the SessionGenerator.